### ISSUE TITLE
TensorQTL fail to calculate the p_beta for h3ack9 qtl

### ISSUE THREAD

**Issue by hsun3163** [cumc/xqtl-pipeline#417]
emprical_pd is the concat of  emprical output for all chromosome generated by:
```
emprical_pd = tibble(map(c('/mnt/vast/hpc/csg/molecular_phenotype_calling/k9_tensorQTL_results/h3k9ac_bed_recipe_h3k9ac_whole.k9_cov.related.filtered.extracted.pca.projected.resid.bed.1.emprical.cis_sumstats.txt','/mnt/vast/hpc/csg/molecular_phenotype_calling/k9_tensorQTL_results/h3k9ac_bed_recipe_h3k9ac_whole.k9_cov.related.filtered.extracted.pca.projected.resid.bed.2.emprical.cis_sumstats.txt','/mnt/vast/hpc/csg/molecular_phenotype_calling/k9_tensorQTL_results/h3k9ac_bed_recipe_h3k9ac_whole.k9_cov.related.filtered.extracted.pca.projected.resid.bed.3.emprical.cis_sumstats.txt','/mnt/vast/hpc/csg/molecular_phenotype_calling/k9_tensorQTL_results/h3k9ac_bed_recipe_h3k9ac_whole.k9_cov.related.filtered.extracted.pca.projected.resid.bed.4.emprical.cis_sumstats.txt','/mnt/vast/hpc/csg/molecular_phenotype_calling/k9_tensorQTL_results/h3k9ac_bed_recipe_h3k9ac_whole.k9_cov.related.filtered.extracted.pca.projected.resid.bed.5.emprical.cis_sumstats.txt','/mnt/vast/hpc/csg/molecular_phenotype_calling/k9_tensorQTL_results/h3k9ac_bed_recipe_h3k9ac_whole.k9_cov.related.filtered.extracted.pca.projected.resid.bed.6.emprical.cis_sumstats.txt','/mnt/vast/hpc/csg/molecular_phenotype_calling/k9_tensorQTL_results/h3k9ac_bed_recipe_h3k9ac_whole.k9_cov.related.filtered.extracted.pca.projected.resid.bed.7.emprical.cis_sumstats.txt','/mnt/vast/hpc/csg/molecular_phenotype_calling/k9_tensorQTL_results/h3k9ac_bed_recipe_h3k9ac_whole.k9_cov.related.filtered.extracted.pca.projected.resid.bed.8.emprical.cis_sumstats.txt','/mnt/vast/hpc/csg/molecular_phenotype_calling/k9_tensorQTL_results/h3k9ac_bed_recipe_h3k9ac_whole.k9_cov.related.filtered.extracted.pca.projected.resid.bed.9.emprical.cis_sumstats.txt','/mnt/vast/hpc/csg/molecular_phenotype_calling/k9_tensorQTL_results/h3k9ac_bed_recipe_h3k9ac_whole.k9_cov.related.filtered.extracted.pca.projected.resid.bed.10.emprical.cis_sumstats.txt','/mnt/vast/hpc/csg/molecular_phenotype_calling/k9_tensorQTL_results/h3k9ac_bed_recipe_h3k9ac_whole.k9_cov.related.filtered.extracted.pca.projected.resid.bed.11.emprical.cis_sumstats.txt','/mnt/vast/hpc/csg/molecular_phenotype_calling/k9_tensorQTL_results/h3k9ac_bed_recipe_h3k9ac_whole.k9_cov.related.filtered.extracted.pca.projected.resid.bed.12.emprical.cis_sumstats.txt','/mnt/vast/hpc/csg/molecular_phenotype_calling/k9_tensorQTL_results/h3k9ac_bed_recipe_h3k9ac_whole.k9_cov.related.filtered.extracted.pca.projected.resid.bed.13.emprical.cis_sumstats.txt','/mnt/vast/hpc/csg/molecular_phenotype_calling/k9_tensorQTL_results/h3k9ac_bed_recipe_h3k9ac_whole.k9_cov.related.filtered.extracted.pca.projected.resid.bed.14.emprical.cis_sumstats.txt','/mnt/vast/hpc/csg/molecular_phenotype_calling/k9_tensorQTL_results/h3k9ac_bed_recipe_h3k9ac_whole.k9_cov.related.filtered.extracted.pca.projected.resid.bed.15.emprical.cis_sumstats.txt','/mnt/vast/hpc/csg/molecular_phenotype_calling/k9_tensorQTL_results/h3k9ac_bed_recipe_h3k9ac_whole.k9_cov.related.filtered.extracted.pca.projected.resid.bed.16.emprical.cis_sumstats.txt','/mnt/vast/hpc/csg/molecular_phenotype_calling/k9_tensorQTL_results/h3k9ac_bed_recipe_h3k9ac_whole.k9_cov.related.filtered.extracted.pca.projected.resid.bed.17.emprical.cis_sumstats.txt','/mnt/vast/hpc/csg/molecular_phenotype_calling/k9_tensorQTL_results/h3k9ac_bed_recipe_h3k9ac_whole.k9_cov.related.filtered.extracted.pca.projected.resid.bed.18.emprical.cis_sumstats.txt','/mnt/vast/hpc/csg/molecular_phenotype_calling/k9_tensorQTL_results/h3k9ac_bed_recipe_h3k9ac_whole.k9_cov.related.filtered.extracted.pca.projected.resid.bed.19.emprical.cis_sumstats.txt','/mnt/vast/hpc/csg/molecular_phenotype_calling/k9_tensorQTL_results/h3k9ac_bed_recipe_h3k9ac_whole.k9_cov.related.filtered.extracted.pca.projected.resid.bed.20.emprical.cis_sumstats.txt','/mnt/vast/hpc/csg/molecular_phenotype_calling/k9_tensorQTL_results/h3k9ac_bed_recipe_h3k9ac_whole.k9_cov.related.filtered.extracted.pca.projected.resid.bed.21.emprical.cis_sumstats.txt','/mnt/vast/hpc/csg/molecular_phenotype_calling/k9_tensorQTL_results/h3k9ac_bed_recipe_h3k9ac_whole.k9_cov.related.filtered.extracted.pca.projected.resid.bed.22.emprical.cis_sumstats.txt'), ~read_delim(.x,"\t")))%>%unnest()


emprical_pd["q_perm"] = qvalue(emprical_pd$p_perm)$qvalue
emprical_pd["fdr_perm"] = p.adjust(emprical_pd$p_perm,"fdr")
summary = tibble("fdr_perm_0.05" =  sum(emprical_pd["fdr_perm"] < 0.05) ,
                  "q_perm_0.05" = sum(emprical_pd["q_perm"] < 0.05) ,
                   "q_perm_0.01" = sum(emprical_pd["q_perm"] < 0.01) )

```
However, none of the peak produce a valid p_beta as demostrated below:

```
nrow(na.omit(emprical_pd["p_beta"]))
nrow(na.omit(emprical_pd["p_perm"]))
nrow((emprical_pd))
```
```
0
90846
90846
```
There are 230 sig peak based on 0.05 of both q_value or fdr

```
sum(emprical_pd["q_perm"] < 0.05)
sum(emprical_pd["fdr_perm"] < 0.05)
```

@Rhopala Please look into it.

**Comment by Rhopala** [comment]
My fault, after some miserable debug I realized I messed up which is the final covariate file after renaming the long A+B output (That could also be the reason why previous TensorQTL run took much more memory than expected). I can get pval_nominal now (which was all NA in previous outputs) so every other calculation base on it including the p_beta mentioned in this post should be fine. Rerunning the tensorQTL now.
![image](https://user-images.githubusercontent.com/78466264/193397999-f8d15a1c-a02c-4122-aff8-24151424421b.png)
